### PR TITLE
Use custom facts to get device names

### PR DESCRIPTION
**Pull request by berendt**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ The following stable releases are supported. The development branch usually work
 ## Test status of cloud providers
 
 * [Betacloud](https://www.betacloud.de): Works
-* [Citycloud](https://www.citycloud.com): Works (need to change disk names from sdX to vdX, pass ``drives_vdx: true`` in environment). You can use ``environment-CityCloud.yml`` to get a set of working parameters.
+* [Citycloud](https://www.citycloud.com): Works
```
Comment (garloff) [inline README.md]: The pointer to environment-CityCloud.yml should be left in (it also specifies flavors, image names, az names that work on CC).

File: environments/custom/files/testbed_ceph_devices.fact [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import json
+import os
+from subprocess import PIPE, run
+
+import requests
+
+response = requests.get('http://169.254.169.254/latest/meta-data/block-device-mapping/root')
+root = os.path.basename(response.text)
+
+devices = run("lsblk -nd -I 8 --output NAME | sort | grep -v %s" % root, stdout=PIPE, stderr=PIPE, universal_newlines=True, shell=True)
```
Comment (garloff) [inline environments/custom/files/testbed_ceph_devices.fact]: Don't limit to SCSI (-I 8). You need to allow sda (8), vda (252), xvda (220) at least: -I 8,220,252.

**Comment by berendt** [comment]
> Testing notes:
> 
> * I did not see these scripts anywhere on the manager node. Is there some image building and syncing cycle required that I need to wait for?

The scripts are not part of the images. You have to change the branch of the configuration repository.

```
$ osism-generic configuration
$ cd /opt/configuration
$ git checkout custom-facts
```

**Comment by berendt** [comment]
@garloff It's getting ready. Can you please test it again. The branch is now configurable.

```
make deploy STACK_PARAMS="--parameter configuration_version=custom-facts"
```

**Comment by berendt** [comment]
Base bootstrap works. Test of Kolla and Ceph follows now.

**Pull request merged**